engine/src/main/java/org/terasology/rendering/logic/MeshRenderer.java

```diff
@@ -244,6 +242,21 @@ private void renderEntitiesByMaterial(SetMultimap<Material, EntityRef> meshByMat
         }
     }
 
+    /**
+     * Checks whether the entity at the given position is relevant.
+     * <p>
+     * The entity is at the given position is relevant if
+     * a) the entity itself is always relevant, or
+     * b) if the block at the position is relevant.
+     *
+     * @param entity   the entity to check for relevance
+     * @param position the position the entity is located
+     * @return true if the entity itself or the block at the given position are relevant, false otherwise.
+     */
+    private boolean isRelevant(EntityRef entity, Vector3f position) {
+        return entity.isAlwaysRelevant() || worldProvider.isBlockRelevant(position);
```

Comment:
Good point! As there should not be too many entities that have to be rendered all the time, the latter is indeed more likely I think.